pkg/code-generator/codegen/templates/snapshot_emitter_test_template.go

```diff
@@ -70,11 +73,14 @@ var _ = Describe("{{ upper_camel .Project.ProjectConfig.Version }}Emitter", func
 		cfg, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
 		Expect(err).NotTo(HaveOccurred())
 
+{{- if $needs_crd_client }}
+
 		cache := kuberc.NewKubeCache()
+{{- end}}
```

Comment:
I think you can just call `NewKubeCache()` on line 92. Since it's needed only there, and given that you land in that block only if `.HasStatus` is true, `needs_crd_client` becomes redundant and can be removed.